lunabot_embedded/firmware/teensy_main/lib/lunabot_drivers/interfaces.cpp

```diff
@@ -136,71 +134,139 @@ volatile uint8_t EncoderBus::spi_buffer_[EncoderBus::BUFFER_SIZE] = {0};
-    }
+  digitalWriteFast(clk_p_,
+                   LOW); // Set CLK line LOW (to inform encoder -> latch data)
+
+  // Before in setup() the pinMode() change the CLK pin function to
+  // output, now we have to enable usage of this pin by SPI port with
+  // calling SPI.begin():
+  SPI.begin();
+  SPI.beginTransaction(spi_settings); // We use transactional API
 
-    SPI.endTransaction(); // We use transactional API
+  for (int i = 0; i < BUFFER_SIZE; i++) {
+    spi_buffer_[i] = SPI.transfer(0xAA); // Transfer anything and read data back
+  }
 
-    pinMode(clk_p_,
-            OUTPUT); // A while before we set CLK pin to be used by SPI
-                     // port, now we have to change it manually...
-    digitalWrite(clk_p_, HIGH); // ... back to idle HIGH
+  SPI.endTransaction(); // We use transactional API
 
-    delayMicroseconds(
-        27); // Running above 500kHz perform Delay First Clock function
+  pinMode(clk_p_,
+          OUTPUT);            // A while before we set CLK pin to be used by SPI
+                              // port, now we have to change it manually...
+  digitalWrite(clk_p_, HIGH); // ... back to idle HIGH
 
-    enc_buffer_[curr_id_] =
-        static_cast<uint32_t>(spi_buffer_[0] << 8) |
-        static_cast<uint32_t>(spi_buffer_[1]); // here transfer8 was made
+  delayMicroseconds(
+      27); // Running above 500kHz perform Delay First Clock function
 
-    curr_id_ = (curr_id_ + 1) % BUS_SIZE;
-    select_enc_(curr_id_);
+  enc_buffer_[curr_id_] =
+      static_cast<uint32_t>(spi_buffer_[0] << 8) |
+      static_cast<uint32_t>(spi_buffer_[1]); // here transfer8 was made
+
+  curr_id_ = (curr_id_ + 1) % BUS_SIZE;
+  select_enc_(curr_id_);
 }
 
 float EncoderBus::read_enc(uint8_t id) {
-    uint32_t data;
-    noInterrupts();
-    data = enc_buffer_[id];
-    interrupts();
+  uint32_t data;
+  noInterrupts();
+  data = enc_buffer_[id];
+  interrupts();
 
-    // Shift one bit right - (MSB of position was placed on at MSB of
-    // uint32_t, so make it right and shift to make MSB position at 30'th
-    // bit): data = data >> 1;
+  // Shift one bit right - (MSB of position was placed on at MSB of
+  // uint32_t, so make it right and shift to make MSB position at 30'th
+  // bit): data = data >> 1;
 
-    // encoderPosition is placed in front (starting from MSB of uint32_t) so
-    // shift it for 11 bits to align position data right
+  // encoderPosition is placed in front (starting from MSB of uint32_t) so
+  // shift it for 11 bits to align position data right
 
-    return static_cast<float>(data) / static_cast<float>(1 << 16) * 360.0F;
+  return static_cast<float>(data) / static_cast<float>(1 << 16) * 360.0F;
+}
+
+volatile float UWBBus::uwb_buffer_[BUS_SIZE] = {0};
+
+float UWBBus::read_uwb(uint8_t id) {
+  noInterrupts();
+  uint8_t data = uwb_buffer_[id];
+  interrupts();
+  return data;
+}
+
+void UWBBus::init() {
+  Serial2.begin(115200);
```

Comment:
Should be Serial8, reference [teensy hardware serial reference table](https://www.pjrc.com/teensy/td_uart.html) and [pcb schematic and footprint](https://docs.google.com/spreadsheets/d/1eX79YtawJqBA8VePFFtKJT6RR21gvK4qH1DKJX_vJE0/edit#gid=596330011)